### PR TITLE
Show personal deadline extensions in ToC view

### LINKED ISSUE #1277: Show personal deadline extensions in ToC view
A student’s points page now displays personal deadline extensions, which is useful. It might make sense to add a similar notification elsewhere in the A+ web UI as well. The table of contents (e.g., https://plus.cs.aalto.fi/o1/2023/) comes to mind, at least, and there might be other places as well.

### PR DESCRIPTION
**Pull request by ashi006**
Fixes #1277

# Description

**What?**

A student’s points page displays personal deadline extensions, which is useful. But it might make sense to add a similar notification elsewhere in the A+ web UI as well.

**How?**

Now, the ToC view also shows labels for personal deadline extensions.

Fixes #1277 

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [x] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [x] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: exercise/templates/exercise/_children.html [hunk 1]
```diff
@@ -15,6 +15,7 @@
 				{{ entry.name|parse_localization }}
 			{% else %}
 				<a href="{{ entry.link }}">{{ entry.name|parse_localization }}</a>
+				{% check_exercise_extensions children entry.name %}
```
Comment (ihalaij1) [inline exercise/templates/exercise/_children.html:18]: Could this template tag be named `check_exercise_deadline_extensions`?
`check_exercise_extension` sounds like something different.

File: exercise/templatetags/exercise.py [hunk 2]
```diff
@@ -73,6 +74,24 @@ def _is_accessible(context, entry, t):
     return True
 
 
+@register.simple_tag
+def check_exercise_extensions(children, entry_name):
+    personal_deadline_printed = False
```
Comment (ihalaij1) [inline exercise/templatetags/exercise.py:79]: Could this template tag be named `check_exercise_deadline_extensions`?
`check_exercise_extension` sounds like something different.

**Review by ihalaij1 — CHANGES_REQUESTED** [review]
Nice!

I wrote one minor comment, and also could this feature be added to the module specific ToC page too? Example:
image

File: exercise/templates/exercise/_children.html (hunk 1 above)
Comment (ashi006) [reply exercise/templates/exercise/_children.html:18]: Yes, I can check.

File: exercise/templatetags/exercise.py (hunk 2 above)
Comment (ashi006) [reply exercise/templatetags/exercise.py:79]: Sure.

**Comment by ashi006** [comment]
Changes made, please review.

**Review by ihalaij1 — APPROVED** [review]
Excellent!

I made some small changes to move the HTML from `exercise.py` to the translation files.

I will merge this to the `master-with-cache` branch, since it is dependent on the cache changes that were moved away from `master`.

**Pull request merged**